platform/android/MapboxGLAndroidSDKTestApp/src/main/java/org/maplibre/android/testapp/activity/annotation/BulkMarkerActivity.kt

```diff
@@ -102,12 +101,25 @@ class BulkMarkerActivity : AppCompatActivity(), OnItemSelectedListener {
         for (i in 0 until amount) {
             randomIndex = random.nextInt(locations!!.size)
             val latLng = locations!![randomIndex]
-            maplibreMap.addSymbol(
-                KSymbol(
+            maplibreMap.addAnnotation(
+                Symbol(
                     position = latLng,
                     text = Text(i.toString(), color = Color.WHITE),
                     icon = icon
-                )
+                ).apply {
+                    if (i == 0) {
+                        clickListener = object : OnSymbolClickListener {
```

Comment:
Could this be Kotlin-ified further by just using a lambda? Or do we want to keep it this way for limited backwards compatibility?

Reply:
Sure can! This is a great idea because it will clean up Kotlin code but not have negative impacts on Java code (provided that lambdas are used). Somehow it will make our code a bit less pretty but that is well worth it.